prometheus/internalmetrics/internal_metrics.go

```diff
@@ -103,6 +104,13 @@ var Metrics = MetricsType{
 		},
 		[]string{},
 	),
+	APIFailures: prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "kiali_api_failures_total",
+			Help: "Counts the total number of failures encountered by a particular API handler.",
+		},
+		[]string{labelRoute},
+	),
```

Comment:
I'm not against this new metric, I'm only wondering why the metric is specific to failures and not just recording all response codes?  Ignore this if it a silly question, I have never really looked at out metrics.   Assuming the new metric is just recording failures, why is it limited (below) to just 500 and 503?  Finally, would it be possible to record the API being called so we could tie together API with responses?

Reply:
> Finally, would it be possible to record the API being called so we could tie together API with responses?

I'm pretty sure this stuff does that - its an attribute on the time series IIRC.

Reply:
@jshaughn like @jmazzitelli said, it does record the API being called.

I'm limiting  the errors to 503 and 500 because I check edthe comments in every method and there wasn't a reference to another 5xx but I can re check that.

This API failure metric came out as an idea to replace the old "failures" that was used to show business failures, however It's possible to add a new dimension (label) to record status codes so that would introduce a new aggregation in the graph to show any status code return by the server. Instead of counting failures, it will count every response's status code (including 4xx, 3xx).

Reply:
I tried to add a new label to the metric: kiali_api_processing_duration_seconds to let users filter API calls by response code but after some testing, I realized I cannot create the timer with the label status code because in that moment of time the status code is not known. 

Anyway, this is not necessarily what we were discussing, I guess the decision is to leave the failure counter as it is right now or extend the counter to count every status code. 

My thoughts are that this counter's objective is to have more information in case of problems, relating the failure with an specfici API route.